easybuild/tools/module_generator.py

```diff
@@ -1513,19 +1536,23 @@ def set_environment(self, key, value, relpath=False):
             self.log.info("Not including statement to define environment variable $%s, as specified", key)
             return ''
 
-        value, use_pushenv = self.unpack_setenv_value(key, value)
+        set_value, use_pushenv, shell_vars = self.unpack_setenv_value(key, value)
 
         if relpath:
-            if value:
-                val = self.PATH_JOIN_TEMPLATE % value
-            else:
-                val = 'root'
+            set_value = self._path_join_cmd(set_value)
+            if shell_vars:
+                set_value = self.REGEX_QUOTE_SHELL_VAR.sub(r'os.getenv("\1")', set_value)
         else:
-            val = quote_str(value)
+            if shell_vars:
+                set_value = self.REGEX_SHELL_VAR.sub(rf'{self.CONCAT_STR}os.getenv("\1"){self.CONCAT_STR}', set_value)
```

Comment:
This also deserves a comment on top with an example of what this does

Reply:
fixed in https://github.com/easybuilders/easybuild-framework/commit/510e8f0942598ec8856ffb5d7fefd0c56fac235a